### ISSUE TITLE
Using the installation notes for Linux on Win10

### ISSUE THREAD

**Issue by Resonanz** [soft-matter/trackpy#475]
This is a really great and easy way of installing trackpy on Linux on Windows 10 (requires a newish build of Win10):

https://github.com/soft-matter/trackpy/wiki/The-Ten-Commandments-for-configuring-trackpy-on-Anaconda-on-Debian-Linux-on-Windows-10-using-WSL

If you install pims or trackpy inside a user-created conda environment rather than in the root environment, then expect errors as I document below. After many attempts to get this right LOL..... my Wiki instructions **should** now work perfectly and be error-free (2018-03-17, trackpy 0.3.3, python 3.6).

UPDATE: OH DAMMIT, THIS STILL FAILS. SO THIS APPEARS TO BE HOW IT WORKS....

CASE 1: INSTALL PIMS AND TRACKPY IN THE ROOT ENV ---> CANNOT IMPORT PIMS OR TRACKPY FROM THE COMMAND LINE INSIDE A CREATED CONDA ENV.

CASE 2: INSTALL PIMS AND TRACKPY IN A CREATED CONDA ENV ---> CANNOT IMPORT PIMS OR TRACKPY INTO JUPYTER NOTEBOOK.

SIGH.

**Comment by Resonanz** [comment]
One failure: `PermissionError(13, 'Permission denied')`

This is the fail (can see below):

certifi 2018.1.18: #####################################################################7 | 75%


**FULL OUTPUT**

(py36) rob@WIN10-ELITEDESK-i7:~$ conda install -c conda-forge trackpy
Solving environment: done


==> WARNING: A newer version of conda exists. <==
  current version: 4.4.10
  latest version: 4.4.11

Please update conda by running

    $ conda update -n base conda



## Package Plan ##

  environment location: /home/rob/anaconda3/envs/py36

  added / updated specs:
    - trackpy


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    ca-certificates-2018.1.18  |                0         140 KB  conda-forge
    lzo-2.10                   |                0         234 KB  conda-forge
    fontconfig-2.12.6          |                0         943 KB  conda-forge
    libxml2-2.9.7              |                0         5.0 MB  conda-forge
    slicerator-0.9.8           |             py_1          10 KB  conda-forge
    bzip2-1.0.6                |                1         310 KB  conda-forge
    jinja2-2.10                |           py36_0         183 KB  conda-forge
    numexpr-2.6.4              |           py36_1         428 KB  conda-forge
    six-1.11.0                 |           py36_1          21 KB  conda-forge
    libxcb-1.12                |                1         454 KB  conda-forge
    libiconv-1.15              |                0         2.0 MB  conda-forge
    python-dateutil-2.7.0      |             py_0         200 KB  conda-forge
    expat-2.2.5                |                0         434 KB  conda-forge
    markupsafe-1.0             |           py36_0          32 KB  conda-forge
    pyqt-5.6.0                 |           py36_4         5.3 MB  conda-forge
    xorg-libxau-1.0.8          |                3          14 KB  conda-forge
    freetype-2.8.1             |                0         2.9 MB  conda-forge
    libgfortran-3.0.0          |                1         281 KB
    pandas-0.22.0              |           py36_0        26.1 MB  conda-forge
    pims-0.4.1                 |             py_1          59 KB  conda-forge
    icu-58.2                   |                0        22.7 MB  conda-forge
    libpng-1.6.34              |                0         305 KB  conda-forge
    pytz-2018.3                |             py_0         217 KB  conda-forge
    matplotlib-2.2.0           |           py36_0         9.2 MB  conda-forge
    pcre-8.41                  |                1         729 KB  conda-forge
    kiwisolver-1.0.1           |           py36_1         925 KB  conda-forge
    olefile-0.45.1             |           py36_0          49 KB  conda-forge
    dbus-1.10.22               |                0         1.6 MB  conda-forge
    certifi-2018.1.18          |           py36_0         143 KB  conda-forge
    sip-4.18                   |           py36_1         263 KB  conda-forge
    jpeg-9b                    |                2         1.0 MB  conda-forge
    backports-1.0              |           py36_1           4 KB  conda-forge
    openssl-1.0.2n             |                0         3.5 MB  conda-forge
    pyparsing-2.2.0            |           py36_0          95 KB  conda-forge
    numpy-1.14.2               |   py36hdbf6ddf_0         4.0 MB
    pillow-5.0.0               |           py36_0         975 KB  conda-forge
    tifffile-0.14.0            |           py36_1         195 KB  conda-forge
    trackpy-0.3.3              |           py36_0         111 KB  conda-forge
    xorg-libxdmcp-1.1.2        |                3          21 KB  conda-forge
    libtiff-4.0.9              |                0         511 KB  conda-forge
    cycler-0.10.0              |           py36_0          14 KB  conda-forge
    pytables-3.4.2             |           py36_7         7.0 MB  conda-forge
    hdf5-1.10.1                |                2         5.0 MB  conda-forge
    backports.functools_lru_cache-1.5|           py36_0          11 KB  conda-forge
    pyyaml-3.12                |           py36_1         394 KB  conda-forge
    libgfortran-ng-7.2.0       |       hdf63c60_3         1.2 MB
    tornado-4.5.3              |           py36_0         628 KB  conda-forge
    yaml-0.1.7                 |                0         302 KB  conda-forge
    ------------------------------------------------------------
                                           Total:       106.2 MB

The following NEW packages will be INSTALLED:

    backports:                     1.0-py36_1            conda-forge
    backports.functools_lru_cache: 1.5-py36_0            conda-forge
    bzip2:                         1.0.6-1               conda-forge
    cycler:                        0.10.0-py36_0         conda-forge
    dbus:                          1.10.22-0             conda-forge
    expat:                         2.2.5-0               conda-forge
    fontconfig:                    2.12.6-0              conda-forge
    freetype:                      2.8.1-0               conda-forge
    glib:                          2.53.6-h5d9569c_2
    gst-plugins-base:              1.12.4-h33fb286_0
    gstreamer:                     1.12.4-hb53b477_0
    hdf5:                          1.10.1-2              conda-forge
    icu:                           58.2-0                conda-forge
    intel-openmp:                  2018.0.0-hc7b2577_8
    jinja2:                        2.10-py36_0           conda-forge
    jpeg:                          9b-2                  conda-forge
    kiwisolver:                    1.0.1-py36_1          conda-forge
    libgfortran:                   3.0.0-1
    libgfortran-ng:                7.2.0-hdf63c60_3
    libiconv:                      1.15-0                conda-forge
    libpng:                        1.6.34-0              conda-forge
    libtiff:                       4.0.9-0               conda-forge
    libxcb:                        1.12-1                conda-forge
    libxml2:                       2.9.7-0               conda-forge
    lzo:                           2.10-0                conda-forge
    markupsafe:                    1.0-py36_0            conda-forge
    matplotlib:                    2.2.0-py36_0          conda-forge
    mkl:                           2018.0.1-h19d6760_4
    numexpr:                       2.6.4-py36_1          conda-forge
    numpy:                         1.14.2-py36hdbf6ddf_0
    olefile:                       0.45.1-py36_0         conda-forge
    pandas:                        0.22.0-py36_0         conda-forge
    pcre:                          8.41-1                conda-forge
    pillow:                        5.0.0-py36_0          conda-forge
    pims:                          0.4.1-py_1            conda-forge
    pyparsing:                     2.2.0-py36_0          conda-forge
    pyqt:                          5.6.0-py36_4          conda-forge
    pytables:                      3.4.2-py36_7          conda-forge
    python-dateutil:               2.7.0-py_0            conda-forge
    pytz:                          2018.3-py_0           conda-forge
    pyyaml:                        3.12-py36_1           conda-forge
    qt:                            5.6.2-h974d657_12
    scipy:                         1.0.0-py36hbf646e7_0
    sip:                           4.18-py36_1           conda-forge
    six:                           1.11.0-py36_1         conda-forge
    slicerator:                    0.9.8-py_1            conda-forge
    tifffile:                      0.14.0-py36_1         conda-forge
    tornado:                       4.5.3-py36_0          conda-forge
    trackpy:                       0.3.3-py36_0          conda-forge
    xorg-libxau:                   1.0.8-3               conda-forge
    xorg-libxdmcp:                 1.1.2-3               conda-forge
    yaml:                          0.1.7-0               conda-forge

The following packages will be UPDATED:

    ca-certificates:               2017.08.26-h1d4fec5_0             --> 2018.1.18-0      conda-forge
    certifi:                       2018.1.18-py36_0                  --> 2018.1.18-py36_0 conda-forge
    openssl:                       1.0.2n-hb7f436b_0                 --> 1.0.2n-0         conda-forge

Proceed ([y]/n)? y


Downloading and Extracting Packages
ca-certificates 2018.1.18: ##################################################################################### | 100%
lzo 2.10: ###################################################################################################### | 100%
fontconfig 2.12.6: ############################################################################################# | 100%
libxml2 2.9.7: ################################################################################################# | 100%
slicerator 0.9.8: ############################################################################################## | 100%
bzip2 1.0.6: ################################################################################################### | 100%
jinja2 2.10: ################################################################################################### | 100%
numexpr 2.6.4: ################################################################################################# | 100%
six 1.11.0: #################################################################################################### | 100%
libxcb 1.12: ################################################################################################### | 100%
libiconv 1.15: ################################################################################################# | 100%
python-dateutil 2.7.0: ######################################################################################### | 100%
expat 2.2.5: ################################################################################################### | 100%
markupsafe 1.0: ################################################################################################ | 100%
pyqt 5.6.0: #################################################################################################### | 100%
xorg-libxau 1.0.8: ############################################################################################# | 100%
freetype 2.8.1: ################################################################################################ | 100%
libgfortran 3.0.0: ############################################################################################# | 100%
pandas 0.22.0: ################################################################################################# | 100%
pims 0.4.1: #################################################################################################### | 100%
icu 58.2: ###################################################################################################### | 100%
libpng 1.6.34: ################################################################################################# | 100%
pytz 2018.3: ################################################################################################### | 100%
matplotlib 2.2.0: ############################################################################################## | 100%
pcre 8.41: ##################################################################################################### | 100%
kiwisolver 1.0.1: ############################################################################################## | 100%
olefile 0.45.1: ################################################################################################ | 100%
dbus 1.10.22: ################################################################################################## | 100%
certifi 2018.1.18: #####################################################################7                        |  75%
sip 4.18: ###################################################################################################### | 100%
jpeg 9b: ####################################################################################################### | 100%
backports 1.0: ################################################################################################# | 100%
openssl 1.0.2n: ################################################################################################ | 100%
pyparsing 2.2.0: ############################################################################################### | 100%
numpy 1.14.2: ################################################################################################## | 100%
pillow 5.0.0: ################################################################################################## | 100%
tifffile 0.14.0: ############################################################################################### | 100%
trackpy 0.3.3: ################################################################################################# | 100%
xorg-libxdmcp 1.1.2: ########################################################################################### | 100%
libtiff 4.0.9: ################################################################################################# | 100%
cycler 0.10.0: ################################################################################################# | 100%
pytables 3.4.2: ################################################################################################ | 100%
hdf5 1.10.1: ################################################################################################### | 100%
backports.functools_lru_cache 1.5: ############################################################################# | 100%
pyyaml 3.12: ################################################################################################### | 100%
libgfortran-ng 7.2.0: ########################################################################################## | 100%
tornado 4.5.3: ################################################################################################# | 100%
yaml 0.1.7: #################################################################################################### | 100%

PermissionError(13, 'Permission denied')

**Comment by Resonanz** [comment]
If I enter my py36 environment using:

`source activate py36`

they type:

`import pims`

and then:

`import trackpy`

then everything appears to be perfect. However running:

`jupyter notebook`

and navigating to the trackpy-examples/notebooks and loading the walkthough example fails to import pims :-(

`---------------------------------------------------------------------------
ModuleNotFoundError                       Traceback (most recent call last)
<ipython-input-2-b3de2f250249> in <module>()
      3 from pandas import DataFrame, Series  # for convenience
      4 
----> 5 import pims
      6 import trackpy as tp

ModuleNotFoundError: No module named 'pims'`

**Comment by Resonanz** [comment]
I also tries to install pims via conda:

`(py36) rob@WIN10-ELITEDESK-i7:~$ conda install -c conda-forge pims
Solving environment: done

# All requested packages already installed.`

But this results in the same failure:

`---------------------------------------------------------------------------
ModuleNotFoundError                       Traceback (most recent call last)
<ipython-input-2-b3de2f250249> in <module>()
      3 from pandas import DataFrame, Series  # for convenience
      4 
----> 5 import pims
      6 import trackpy as tp

ModuleNotFoundError: No module named 'pims'`

**Comment by Resonanz** [comment]
So I will try to install pims in the "root" environment:

source activate root

But it wants to downgrade conda :-(

`(base) rob@WIN10-ELITEDESK-i7:~$ conda install -c conda-forge pims
Solving environment: done

## Package Plan ##

  environment location: /home/rob/anaconda3

  added / updated specs:
    - pims


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    conda-4.3.34               |           py36_0         514 KB  conda-forge
    anaconda-custom            |   py36hbbc8b67_0           6 KB
    ------------------------------------------------------------
                                           Total:         520 KB

The following NEW packages will be INSTALLED:

    pims:            0.4.1-py_1            conda-forge
    slicerator:      0.9.8-py_1            conda-forge
    tifffile:        0.14.0-py36_1         conda-forge

The following packages will be UPDATED:

    anaconda:        5.1.0-py36_2                      --> custom-py36hbbc8b67_0
    ca-certificates: 2017.08.26-h1d4fec5_0             --> 2018.1.18-0           conda-forge
    certifi:         2018.1.18-py36_0                  --> 2018.1.18-py36_0      conda-forge
    openssl:         1.0.2n-hb7f436b_0                 --> 1.0.2n-0              conda-forge

The following packages will be DOWNGRADED:

    conda:           4.4.11-py36_0                     --> 4.3.34-py36_0         conda-forge

Proceed ([y]/n)?`

**Comment by Resonanz** [comment]
Let's try instead:

`pip install pims`

Here goes:

`(base) rob@WIN10-ELITEDESK-i7:~$ pip install pims
Collecting pims
  Using cached PIMS-0.4.1-py3-none-any.whl
Requirement already satisfied: six>=1.8 in ./anaconda3/lib/python3.6/site-packages (from pims)
Requirement already satisfied: numpy>=1.7 in ./anaconda3/lib/python3.6/site-packages (from pims)
Collecting slicerator>=0.9.7 (from pims)
Installing collected packages: slicerator, pims
Successfully installed pims-0.4.1 slicerator-0.9.8`

**Comment by Resonanz** [comment]
Sigh :-(  Now trackpy cannot be found......

`(base) rob@WIN10-ELITEDESK-i7:~$ source activate py36
(py36) rob@WIN10-ELITEDESK-i7:~$ jupyter notebook`

Then...

`---------------------------------------------------------------------------
ModuleNotFoundError                       Traceback (most recent call last)
<ipython-input-2-b3de2f250249> in <module>()
      4 
      5 import pims
----> 6 import trackpy as tp

ModuleNotFoundError: No module named 'trackpy'`

**Comment by Resonanz** [comment]
So going back to the root env and conda installing trackpy fixes things..... so it seems that trackpy and pims need to be installed in the root env, which seems rather odd, or is this a bug caused by a path mixup during the install?

**Comment by caspervdw** [comment]
Hi @Resonanz, I looked through your posted logs and it seems to me that it should have worked in the custom environment. The thing is that jupyter notebook does not necessarily take the active conda env as the default python kernel. You have to register your environment with jupyter by installing nb_conda in the enironment. Then choose the right kernel from jupyter notebook.

This should go into the How to install trackpy docs.

Either way, this is not a trackpy issue, so closing this.

**Comment by caspervdw** [comment]
@Resonanz  Great that you posted your detailed installation notes of trackpy in a Debian VM on a Win10! I am however not sure how useful it would be to trackpy users -- the trackpy installation part is already covered by the existing wiki (which I updated just now). If you have your VM up and running, you can use the Anaconda installation guide to get conda, and then use the already existing installation notes to get trackpy.

Anyway, having two installation manuals is not good. Maybe you can move your added wiki page to your own repository? I tried to get the trackpy-specific parts from your notes into the existing notes.

Could you check if the adapted notes work for you now? I am happy to help (and update the notes) if there are still issues.

**Comment by Resonanz** [comment]
Thanks Casper. I was a little excited to be able to get trackpy installed using WSL, so I installed Anaconda and set up a separate conda environment to install trackpy. But after spending most of yesterday trying various combinations to get it working inside the conda environment, I went back to the standard install instructions and noticed that no special environment is used at all. This seemed a little weird. I really think that, if using separate environments is the *proper* way to do things, that this should be in the standard trackpy install instructions. So how can we move forward on this?  I can test the installation instructions, but my python skills are insufficient to configure the innards which is probably required.
